Flabbergasted-Flamingos/bangazon-commerce-app
# Cd category

# Description
Added a category button on the nav bar.  The button takes you to a category list, where you can select a category and see all items in that category.

## Number of Fixes
Type of number of fixes included in this PR.

## Related Ticket(s)
#8 #9

## Problem to Solve
Select products by category.

## Proposed Changes
Describe the proposed changes.

## Expected Behavior
Describe the expected behavior.

## Steps to Test Solution
```
git fetch --all
git checkout cd_category
rm -r db.sqlite3
rm -r website/migrations
python manage.py makemigrations website
python manage.py migrate
python manage.py runserver
```
```
INSERT INTO auth_user VALUES (null, "f", 2018/8/18, 0, "ffff2", "ffff", "f@f.com", 1, 1, 2018/7/15, 'ffff');
INSERT INTO auth_user VALUES (null, "G", 2018/8/18, 0, "ggg2", "ggggg", "g@g.com",  0, 1, 2018/7/15,'ggg');
INSERT INTO website_paymentcategory VALUES (null,  null, 'Visa');
INSERT INTO website_paymentcategory VALUES (null, null, 'MasterCard');
INSERT INTO website_paymentcategory VALUES (null, null, 'AMEX');
INSERT INTO website_producttype VALUES (null, null, 'Home Goods');
INSERT INTO website_producttype VALUES (null, null, 'Pets');
INSERT INTO website_product VALUES (null, null, '', 'Poop-a-Rie', 12, "Make your poop not stink", '1',"Nashville", 1, 2);
INSERT INTO website_product VALUES (null, null, '', 'Febreeze', 12, "Make your house not stink", '1', "Nashville", 1, 2);
INSERT INTO website_product VALUES (null, null, '', 'Sandalwood Candle', 12, "Make your house like a delicious forest", '1', "Knoxville", 1, 1);
INSERT INTO website_product VALUES (null, null, '', 'Puppies', 12, "Australian Shepherd babies", '6', "Knoxville", 2, 2);
INSERT INTO website_product VALUES (null, null, '', 'Sugar Baby', 12, "Sweet Sugary Baby goodness", "Knoxville", '2', 2, 1);
INSERT INTO website_product VALUES (null, null, '' , 'Hamster', 12, "Furry cute rodents", '1', "Knoxville", 2, 1);
```
1. Go to page. Click on the categories button on the navbar.  It should take you to a list view of all of the product categories.

2. Click on one of the product categories.  It should take you to a product list view of all of the products in that category.

3. Click on one of the products.  It should take you to the product detail page.


## Testing

[ ] There are new unit tests in this PR, and I verify that there is full coverage of all new code.
[ ] I certify that all existing tests pass

## Documentation
[ ] There is new documentation in this pull request that must be reviewed..
[ ] I added documentation for any new classes/methods

## Deploy Notes
Notes regarding deployment the contained body of work.  These should note any
db migrations, etc.
